cmd/state-installer/cmd.go

```diff
@@ -210,14 +212,34 @@ func main() {
 			an.EventWithLabel(AnalyticsCat, "error", errs.JoinMessage(err))
 			logging.Critical("Installer error: " + errs.JoinMessage(err))
 		}
+
 		an.EventWithLabel(AnalyticsFunnelCat, "fail", err.Error())
+		// if error carries an exit code, set that and skip error logging
+		if exCode, ok := extractExitCode(err); ok {
+			exitCode = exCode
+			return
+		}
```

Comment:
You didn't achieve anything by creating this function, you're still exiting 1 even when the error didn't satisfy the interface. Please just use `errs.UnwrapExitCode`.

I think what you want is some way to communicate "Exit code came from subprocess". There's merit there but let's not get distracted in this story. Feel free to file a followup.

Reply:
Added as a comment to https://activestatef.atlassian.net/browse/DX-163